components/throttle.js

```diff
@@ -0,0 +1,93 @@
+// throttle.js
+
+var _ = require('underscore');
+var fs = require('fs');
+
+var idIndex = 0;
+var ids;
+
+var logger = require('../src/logger.js');
+var wrapper = require('../src/split-wrapper.js');
+
+/** 
+ * Reads a file with one record per line, and outputs a hash with the 
+ * throttle_size number of entries for processing downstream. For example, 
+ * given a file with values 1, 2, 3, 4 and a throttle size of 2, this 
+ * component would first output a hash that reads {"1": "1", "2": "2"}.  If a 
+ * subsequent throttle size packet of 1 comes in, it would then send 
+ * {"3": "3"}.  One more packet of size 1 would send {"4": "4"}.
+ *
+ * This component is typically placed at the front of a pipeline to 
+ * avoid flooding the pipeline with too many requests.  It is most often
+ * used with the throttle-fire component at the end of the pipeline to 
+ * notify the throttle when it's time to send the next batch of elements
+ * on for processing.
+ * 
+ * @this vni
+ *
+ * @param throttle_size - the number of hash values to be sent 
+ * @param file_envvar an environment variable with the path to the file
+ *                    to be processed.
+ * @param encoding - file encoding; defaults to UTF-8
+ *
+ * @return a hash with the next elements to be processed.
+ */
+module.exports = wrapper(function(throttle_size, file_envvar, encoding) {
```

Comment:
SendNext (and Kick) are about firing off and doing the exact same thing.  There is no setting on them to let you send a parameter to the next component.  My use case is not to just kick it or send a single packet like send next does.    The use case here is to start with an initial large throttle_size and then switch to a smaller size as we finish up.  This approach gives us flexibility for dynamically controlling how much we feed into the pipeline instead of just blindly always sending one.